### DISCUSSION TITLE
USB MIDI / IDE

### DISCUSSION THREAD

**Discussion by Marsupio** [lathoub/Arduino-USBMIDI#19] [Q&A] [answered]
Probably i am missing something, but since the Arduino is set to be recognized as a MIDI device, how you set it back to the default for IDE programming? Are you supposed to change something, or the IDE just keeps on seeing the device as before? (Last time i did it, I had to do bothersome stuff as flashing bootloader!)

**Comment by Marsupio** [comment, marked as answer]
Since I got no reply (i had to find the info elsewhere),  i will publish a reply that might be useful for other people.
Arduino Uno and Mega can be made MIDI class compliant so they are recognised as MIDI device. But it is necessary to flash their secondary chip (through ICSP pins) each time you need to switch them between IDE programming use or MIDI device use, which is not practical at all.
Arduino Leonardo and Micro (ATMega32u4 chip) do not need that process to be made, so you can just program and test your projects at the same time using this library.

**Reply by Petrus125** [reply]
Hello, I'm working over similar boards, and as far I get, Arduino Mega and Uno with Atmega16U2 (serial to usb conv IC), both can be flashed with Mocolufa fw so they get MidiClass Compliant. And works perfect, I tested. Not so bad. Once you got them with the right code, just add the jumper. With a small switch also does the work. Simple and efective. And the power of the Mega is amazing for multiple ins/outs, serials, etc. 
Leonardo and ProMicro, do not need this flash mod. Both are seen as Midi device, ( no drivers, no flash, )and can be programmed directly. Remember to use the correct libs, as they need different ones.
